src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py

```diff
@@ -2102,9 +2102,18 @@ def _bool_reduce_helper(
         assert agg_func in ("all", "any")
 
         frame = self._modin_frame
+
+        # Even though it incurs an extra query, we must get the length of the index to prevent errors.
+        # Empty Series objects return True/False for all/any respectively, regardless of the Series dtype.
+        # Otherwise, Series([], dtype="object").all() would raise a NotImplementedError while
+        # Series([], dtype=bool).all() would work fine; in reality they both produce the same result, True.
+        empty_index = self.get_axis_len(axis=0) == 0
```

Comment:
I fixed the bug where empty Series worked incorrectly with all/any. None of this logic is needed anymore. We could get rid of `empty_index = self.get_axis_len(axis=0) == 0` but then empty Series of non-bool/int type would raise a NotImplementedError. However, this would reduce the query count everywhere by 1.

Reply:
I got rid of this query - now `pd.Series([], dtype=bool)` works with all/any but `pd.Series([], dtype=float)` doesn't. One reason I did this is because it's inconvenient to modify lots of the existing test query counts where it's expecting only one query.